### ISSUE TITLE
Carthage build is failing

### ISSUE THREAD

**Issue by JetForMe** [PromiseKit/Alamofire-#37]
I'm trying to use PromiseKit & Alamofire with Xcode 10.2/Swift 5. My Cartfile:

```
github "Alamofire/Alamofire" ~> 4.0
github "mxcl/PromiseKit" ~> 6.0
github "PromiseKit/Alamofire-" ~> 3.0
```

When I run `carthage update --platforme ios`, I get:

```
*** Fetching Alamofire-
*** Fetching analytics-ios
*** Fetching PromiseKit
*** Checking out PromiseKit at "6.8.4"
*** Checking out Alamofire at "4.8.2"
*** Checking out Alamofire- at "3.2.2"
*** Building scheme "Alamofire iOS" in Alamofire.xcworkspace
*** Building scheme "PromiseKit" in PromiseKit.xcodeproj
*** Building scheme "PMKAlamofire" in PMKAlamofire.xcodeproj
Build Failed
	Task failed with exit code 65:
	/usr/bin/xcrun xcodebuild -project /Users/me/Projects/Clients/MyClient/repo/Carthage/Checkouts/Alamofire-/PMKAlamofire.xcodeproj -scheme PMKAlamofire -configuration Release -derivedDataPath /Users/me/Library/Caches/org.carthage.CarthageKit/DerivedData/10.2_10E125/Alamofire-/3.2.2 -sdk iphoneos ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/16/jfb809_s2fz01ql7k494f6pc0000gn/T/Alamofire- SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in /Users/me/Projects/Clients/MyClient/repo/Carthage/Checkouts/Alamofire-)

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/16/jfb809_s2fz01ql7k494f6pc0000gn/T/carthage-xcodebuild.E91PEu.log
```

The error in the build log:

```
/Users/me/Projects/Clients/MyClient/repo/Carthage/Checkouts/PromiseKit/Sources/AnyPromise.m:4:13: fatal error: 'PromiseKit/PromiseKit-Swift.h' file not found
    #import <PromiseKit/PromiseKit-Swift.h>
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

By the way, what's the trailing dash in the Cartfile entry for ("Alamofire-")?

**Comment by JetForMe** [comment]
I think this is due to PMKAlamofire building with Swift 3, which isn't supported in Xcode 10.2. Trying `travis-swift-5`

**Comment by mxcl** [comment]
Yes needs to be set to 4.0. Apologies. Did this some of others

**Comment by JetForMe** [comment]
I'm having other issues with the example in the readme, documented here: https://stackoverflow.com/questions/55388281/promisekit-6-alamofire-xcode-10-2-swift-5

**Comment by mxcl** [comment]
https://github.com/PromiseKit/PMKAlamofire/pull/38

**Comment by mxcl** [comment]
Your other issue is a common issue documented in our Troubleshooting guide.

**Comment by mxcl** [comment]
Should be fixed, please update.

> By the way, what's the trailing dash in the Cartfile entry for ("Alamofire-")?

The - because Carthage has a single namespace, so otherwise our Alamofire conflicts with the actual Alamofire and the build aborts.

**Comment by JetForMe** [comment]
After figuring out I had to specify "master" in the Cartfile or it would pick up tag fix-6.5.3, it builds. Still haven't been able to resolve the compile issue without being very explicit about things, and it seems Swift 5 doesn't allow empty in expressions. But that's a separate issue from this. Thanks!

**Comment by mxcl** [comment]
You should not have to specify `master` and generally this is not a good idea.

**Comment by JetForMe** [comment]
If I leave the version in Carthage unspecified, it chooses `fix-6.5.3`. I don't know why, but I assume that's the latest tag? (I couldn't find a way to get Carthage to output a dependency graph).

**Comment by mxcl** [comment]
Leaving the version unspecified is generally not a good idea. Specify `~> 3`, this is standard practice and is what we say to do in our README.

I suggest reading the Carthage documentation before shooting yourself in the foot.
